evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -1235,6 +866,7 @@ def test_automl_search_sampler_ratio(
         problem_type=problem_type,
         sampler_method="auto",
         sampler_balanced_ratio=sampling_ratio,
+        _automl_algorithm="iterative",
```

Comment:
The problem with this test is that `allowed_pipelines` isn't generated until `search()` is run for `DefaultAlgorithm`. However, the data for this test will actually error out on `search()`. This should be a good candidate for a mock algorithm.